frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/validation-utils.ts

```diff
@@ -494,3 +494,118 @@ export const validationSchema = () =>
     },
   });
+
+const resourceDefinitionLite = () => {
+  return yup.array().of(
+    yup.object({
+      name: yup.string(),
+      resource: yup.string(),
+    }),
+  );
+};
+
+const taskValidationSchemaLite = () => {
+  return yup.array().of(
+    yup
+      .object({
+        name: nameValidationSchema((tKey) => i18n.t(tKey)),
+        taskRef: yup
+          .object({
+            name: yup.string(),
+            kind: yup.string(),
+          })
+          .default(undefined),
+        taskSpec: yup.object(),
+        runAfter: yup.array().of(yup.string()),
+        params: yup.array().of(
+          yup.object({
+            name: yup.string(),
+            value: yup.lazy((value) => {
+              if (Array.isArray(value)) {
+                return yup.array().of(yup.string());
+              }
+              return yup.string();
+            }),
+          }),
+        ),
+        resources: yup.object({
+          inputs: resourceDefinitionLite(),
+          outputs: resourceDefinitionLite(),
+        }),
+        when: yup.array().of(
+          yup.object({
+            input: yup.string(),
+            operator: yup.string(),
+            values: yup.array(),
+          }),
+        ),
+
+        workspaces: yup.array().of(
+          yup.object({
+            name: yup.string(),
+            workspace: yup.string(),
+          }),
+        ),
+      })
+      .test(
+        'taskRef-or-taskSpec',
+        i18n.t('pipelines-plugin~TaskSpec or TaskRef must be provided.'),
+        function(task: PipelineTask) {
+          return !!task.taskRef || !!task.taskSpec;
+        },
+      ),
+  );
+};
+
+export const pipelineBuilderFormSchemaLite = (formValues: PipelineBuilderFormYamlValues) => {
+  return yup.object({
+    name: yup.string(),
+    params: yup.array().of(
+      yup.object({
+        name: yup.string(),
+        description: yup.string(),
+        default: yup.string(),
+      }),
+    ),
+    resources: yup.array().of(
+      yup.object({
+        name: yup.string(),
+        type: yup.string(),
+      }),
+    ),
+    workspaces: yup.array().of(
+      yup.object({
+        name: yup.string(),
+      }),
+    ),
+    tasks: taskValidationSchemaLite(),
+    finallyTasks: taskValidationSchemaLite(),
+    listTasks: yup.array().of(
+      yup.object({
+        name: yup.string(),
+        runAfter: validRunAfter(formValues),
+      }),
+    ),
+    finallyListTasks: yup.array().of(
+      yup.object({
+        name: yup.string(),
+      }),
+    ),
+  });
+};
+
+export const validationSchemaLite = () =>
+  yup.mixed().test({
+    test(formValues: PipelineBuilderFormYamlValues) {
+      const formYamlDefinition = yup.object({
+        editorType: yup.string().oneOf(Object.values(EditorType)),
+        yamlData: yup.string(),
+        formData: yup.mixed().when('editorType', {
+          is: EditorType.Form,
+          then: pipelineBuilderFormSchemaLite(formValues),
+        }),
```

Comment:
Small nit: I think you just need to test formData. Arguably you don't even need to nest any of this schema and focus on the formData portion of the schema. Calling it with `newFormData` instead of wrapping `newFormData`. In reality we are not live checking, we just want to validate a specific chunk so we can design this to just do that.

Reply:
done